alerting/src/main/kotlin/org/opensearch/alerting/MonitorMetadataService.kt

```diff
@@ -110,18 +114,56 @@ object MonitorMetadataService :
         }
     }
 
+    @Suppress("ComplexMethod", "ReturnCount")
+    suspend fun upsertWorkflowMetadata(metadata: WorkflowMetadata, updating: Boolean): WorkflowMetadata {
```

Comment:
can we have workflowmetadataservice

Reply:
We can have - I thought that it makes sense to be on one place - but yeah since we already separated the workflow in completely separate stream why not.